src/shelter/shelter.ts

```diff
@@ -89,17 +116,52 @@ export class Shelter {
     );
   };
 
+  static createImportedAccountWithSeed$ = (
+    name: string,
+    password: string | undefined,
+    derivationPath: string,
+    seedPhrase: string
+  ) => {
+    const seed = mnemonicToSeedSync(seedPhrase, password);
+    const privateKey = seedToHDPrivateKey(seed, derivationPath);
+
+    return getPublicKeyAndHash$(privateKey).pipe(
+      switchMap(([publicKey, publicKeyHash]) =>
+        Shelter.saveSensitiveData$({
+          seedPhrase,
+          [publicKeyHash]: privateKey
+        }).pipe(
+          mapTo({
+            name,
+            type: AccountTypeEnum.IMPORTED_ACCOUNT,
+            publicKey,
+            publicKeyHash
+          })
+        )
+      ),
+      catchError(() => {
+        showErrorToast({
+          title: 'Failed to import account.',
+          description: 'This may happen because provided Key is invalid.'
+        });
+
+        return of(undefined);
+      })
+    );
+  };
+
```

Comment:
a lot of duplication.
create 1 method that will import Imported account (for both cases) and just prepare correct data for seed case